### PR TITLE
[RW-9089][risk:no] Test for new user account creation

### PR DESCRIPTION
**Pull request by dmohs**
This includes a test for the new demographics survey.

---
**PR checklist**

- [x] This PR meets the Acceptance Criteria in the JIRA story
- [x] The JIRA story has been moved to Dev Review
- [x] This PR includes appropriate unit tests
- [x] I have added explanatory comments where the logic is not obvious
- [ ] I have run and tested this change locally, and my testing process is described here
- [ ] If this includes a new feature flag, I have created and linked new JIRA tickets to (a) turn on the feature flag and (b) remove it later
- [ ] If this includes an API change, I have run the E2E tests on this change against my local server with [yarn test-local](https://github.com/all-of-us/workbench/blob/master/e2e/README.md#examples) because this PR won't be covered by the CircleCI tests 
- [ ] If this includes a UI change, I have taken screen recordings or screenshots of the new behavior and notified the PO and UX designer in [Slack](https://pmi-engteam.slack.com/archives/C02MWP2RN5P)
- [ ] If this change impacts deployment safety (e.g. removing/altering APIs which are in use) I have documented these in the description
- [ ] If this includes an API change, I have updated the appropriate Swagger definitions and updated the appropriate API consumers
  * AoU UI
  * [Perf tests](https://github.com/broadinstitute/mcnulty/blob/develop/src/test/scala/services/AoU.scala)
  * [Researcher Directory export](https://github.com/all-of-us/workbench/wiki/Researcher-Directory-(RDR-export))
  * Cron tasks - for Offline*Controllers
  * SumoLogic - for EgressAlert

### CODE REVIEW

File: api-proxy/src/reqhandler.mjs [hunk 1]
```diff
@@ -40,8 +40,7 @@ const record = async (req, res, resBody) => {
   // so related URLs are sorted together alphabetically.
   const fname =
     req.index.toString().padStart(2, '0')+' '
-    req.url.pathname.replace(/[/]/g, '|')+req.url.search+'.'+req.method.toLowerCase()
-    +'.mjs'
+    +req.url.pathname.replace(/[/]/g, '|')+req.url.search+'.'+req.method.toLowerCase()+'.mjs'
```
Comment (dmohs) [inline api-proxy/src/reqhandler.mjs:43]: Unrelated bug fix.

File: ui/src/app/components/terms-of-service.tsx [hunk 2]
```diff
@@ -62,6 +62,11 @@ export class TermsOfService extends React.Component<TosProps, TosState> {
       hasAckedAgreement: props.afterPrev,
       hasClickedNext: false,
     };
+    // This bypasses the scroll detection. A better implementation may be to provide a method
+    // on HtmlViewer to force the scroll, thus exercising the scroll detection.
+    (window as any).forceHasReadEntireAgreement = () => {
```
Comment (jmthibault79) [inline ui/src/app/components/terms-of-service.tsx:67]: I put on my hacker hat and tried to use this to let me continue on the TOS screen without doing a proper scroll.  It worked.

But do we care?  I think no.

File: e2ev2/tests/user-create.test.js [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+  await page.evaluate(() => window.forceHasReadEntireAgreement())
+  await expect(isAckCheckboxDisabled()).resolves.toBe(false)
+  await jsClick(page, ackCheckboxSel)
+  await page.click('[role="button"]') // Next button
+
+  await page.waitForSelector('.p-dropdown') // Institution Page
+  await page.click('.p-dropdown')
+  await page.waitForSelector('[role="option"][aria-label="Broad Institute"]')
+  await page.click('[role="option"][aria-label="Broad Institute"]')
+  // Email needs to be real or it will bounce and account creation will fail.
```
Comment (jmthibault79) [inline e2ev2/tests/user-create.test.js:37]: well, matching `@broadinstitute.org` anyway

File: e2ev2/tests/user-create.test.js [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+  await page.goto(config.urlRoot(), {waitUntil: 'networkidle0'})
+  await jsClick(page, '[data-test-id="login"]>div+div>[role="button"]') // Create Account
+
+  await page.waitForSelector(iframeSel) // Agreement Page
+  await page.evaluate(() => window.forceHasReadEntireAgreement())
+  await expect(isAckCheckboxDisabled()).resolves.toBe(false)
+  await jsClick(page, ackCheckboxSel)
+  await page.click('[role="button"]') // Next button
+
+  await page.waitForSelector('.p-dropdown') // Institution Page
```
Comment (jmthibault79) [inline e2ev2/tests/user-create.test.js:33]: I'm seeing a lot of `waitForSelector()` then `click()` with the same arg, so I'm inclined to add a helper function.

**Review by jmthibault79 — APPROVED** [review]

File: e2ev2/tests/user-create.test.js (hunk 3 above)
Comment (dmohs) [reply e2ev2/tests/user-create.test.js:37]: I tried `fake@broadinstitute.org` and could not create an account. The logs showed email bounces.

File: e2ev2/tests/user-create.test.js (hunk 4 above)
Comment (dmohs) [reply e2ev2/tests/user-create.test.js:33]: Oh good point. I'm going to merge this as-is to avoid some churn, but that's the first thing I'll look at in my follow-up branch. I want to give it some thought, since the wait and click are kind of two different things.

File: ui/src/app/pages/login/sign-in.tsx [hunk 5]
```diff
@@ -442,6 +442,7 @@ export class SignInImpl extends React.Component<SignInProps, SignInState> {
 
   private captureCaptchaResponse(token) {
     this.setState({ captchaToken: token, captcha: true });
+    window.dispatchEvent(new Event('captchaSolved'));
```
[the remark this answers is not in the document]
Comment (dmohs) [reply ui/src/app/pages/login/sign-in.tsx:445]: My instinct was to make a utility for this because I do expect us to use them fairly often. They are a fundamental mechanism in this environment.

File: ui/src/app/components/terms-of-service.tsx (hunk 2 above)
Comment (dmohs) [reply ui/src/app/components/terms-of-service.tsx:67]: Indeed. We would want to be cautious around a captcha override, for example.
